src/main/clojure/clara/rules/engine.cljc

```diff
@@ -29,12 +29,12 @@
 ;; Token with no bindings, used as the root of beta nodes.
 (def empty-token (->Token [] {}))
 
-;; Record indicating the negation existing in the working memory.
-(defrecord NegationResult [gen-rule-name])
+;; A marker interface to identify internal facts.
+(definterface ISystemFact)
```

Comment:
Using `defprotocol` for its interface is technically using an implementation detail on the JVM.  It isn't their intended purpose.  They just happen to make an interface for efficiency in some cases.

Reader conditionals seems reasonable.  Just use `isa?` in cljs perhaps.